### PR TITLE
implementation of the task

### PR DESCRIPTION
**Pull request by Ketch92**

### CODE REVIEW

**Review by mfedenko09 — APPROVED** [review]

File: src/main/java/core/dao/ManufacturerDao.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+public interface ManufacturerDao {
+    Manufacturer add(Manufacturer manufacturer);
+    
+    Optional<Manufacturer> get(Long id);
+    
+    List<Manufacturer> getAll();
+    
+    Manufacturer update(Long id, Manufacturer manufacturer);
+
+    Manufacturer delete(Long id);
```
Comment (RomanDubovskyi) [inline src/main/java/core/dao/ManufacturerDao.java]:
```suggestion
    boolean delete(Long id);
```

File: src/main/java/core/dao/ManufacturerDao.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import java.util.Optional;
+
+public interface ManufacturerDao {
+    Manufacturer add(Manufacturer manufacturer);
+    
+    Optional<Manufacturer> get(Long id);
+    
+    List<Manufacturer> getAll();
+    
+    Manufacturer update(Long id, Manufacturer manufacturer);
```
Comment (RomanDubovskyi) [inline src/main/java/core/dao/ManufacturerDao.java]:
```suggestion
    Manufacturer update(Manufacturer manufacturer);
```
id is present in the model

File: src/main/java/core/service/ManufacturerService.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package core.service;
+
+import core.model.Manufacturer;
+import java.util.List;
+
+public interface ManufacturerService {
```
Comment (RomanDubovskyi) [inline src/main/java/core/service/ManufacturerService.java:6]: same here with method signature, stick to the examples from task description

File: src/main/java/core/storage/Storage.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
+package core.storage;
+
+import core.model.Manufacturer;
+import java.util.HashMap;
+import java.util.Map;
+
+public class Storage {
+    public static final Map<Long, Manufacturer> storage = new HashMap<>();
```
Comment (RomanDubovskyi) [inline src/main/java/core/storage/Storage.java]:
```suggestion
    public static final Map<Long, Manufacturer> manufacturers = new HashMap<>();
```
there will be cars and drivers as well

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

File: src/main/java/core/dao/ManufacturerDao.java (hunk 1 above)
Comment (Ketch92) [reply src/main/java/core/dao/ManufacturerDao.java]: Ok

File: src/main/java/core/dao/ManufacturerDao.java (hunk 2 above)
Comment (Ketch92) [reply src/main/java/core/dao/ManufacturerDao.java]: ok, i see

File: src/main/java/core/service/ManufacturerService.java (hunk 3 above)
Comment (Ketch92) [reply src/main/java/core/service/ManufacturerService.java:6]: changed

File: src/main/java/core/storage/Storage.java (hunk 4 above)
Comment (Ketch92) [reply src/main/java/core/storage/Storage.java]: Ok

File: src/main/java/core/dao/ManufacturerDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    @Override
+    public List<Manufacturer> getAll() {
+        return new ArrayList<>(Storage.manufacturersStorage.values());
+    }
+    
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        Manufacturer oldValue = Storage.manufacturersStorage.get(manufacturer.getId());
+        Storage.manufacturersStorage.put(manufacturer.getId(), manufacturer);
+        return oldValue;
```
Comment (RomanDubovskyi) [inline src/main/java/core/dao/ManufacturerDaoImpl.java:32]: we usually don't work with an old value but it's up to you

**Review by RomanDubovskyi — APPROVED** [review]

File: src/main/java/core/dao/ManufacturerDaoImpl.java (hunk 5 above)
Comment (Ketch92) [reply src/main/java/core/dao/ManufacturerDaoImpl.java:32]: I thought of this, but IDK what manufacturer to return, as there is no point to return the one we are passing into the method.

**Pull request merged**